src/m365/entra/commands/group/group-set.ts

```diff
@@ -0,0 +1,330 @@
+    );
+  }
+
+  #initTypes(): void {
+    this.types.string.push('id', 'displayName', 'newDisplayName', 'description', 'mailNickname', 'ownerIds', 'ownerUserNames', 'memberIds', 'memberUserNames', 'visibility');
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    let groupId = args.options.id;
+    let ownerIds: string[] = [];
+    let memberIds: string[] = [];
+
+    try {
+      if (args.options.displayName) {
+        if (this.verbose) {
+          await logger.logToStderr(`Retrieving group id...`);
+        }
+
+        groupId = await entraGroup.getGroupIdByDisplayName(args.options.displayName);
+      }
+
+      const requestBody = this.createRequestBody(args.options);
+      const requestOptions: CliRequestOptions = {
+        url: `${this.resource}/v1.0/groups/${groupId}`,
+        headers: {
+          accept: 'application/json;odata.metadata=none'
+        },
+        data: requestBody
+      };
+
+      await request.patch(requestOptions);
+
+      ownerIds = await this.getUserIds(logger, args.options.ownerIds, args.options.ownerUserNames);
+      if (ownerIds.length !== 0) {
+        await this.updateUsers(groupId!, 'owners', ownerIds);
+      }
+      else if (this.verbose) {
+        await logger.logToStderr(`No owners to update.`);
+      }
+
+      memberIds = await this.getUserIds(logger, args.options.memberIds, args.options.memberUserNames);
+      if (memberIds.length !== 0) {
+        await this.updateUsers(groupId!, 'members', memberIds);
+      }
+      else if (this.verbose) {
+        await logger.logToStderr(`No members to update.`);
+      }
+    }
+    catch (err: any) {
+      this.handleRejectedODataJsonPromise(err);
+    }
+  };
+
+  private createRequestBody(options: Options): any {
+    const requestBody: any = {
+      displayName: options.newDisplayName,
+      description: options.description === '' ? null : options.description,
+      mailNickName: options.mailNickname,
+      visibility: options.visibility
+    };
+
+    this.addUnknownOptionsToPayload(requestBody, options);
+    return requestBody;
+  }
+
+  private async getUserIds(logger: Logger, userIds?: string, userNames?: string): Promise<string[]> {
+    if (userIds) {
+      return formatting.splitAndTrim(userIds);
+    }
+
+    if (userNames) {
+      if (this.verbose) {
+        await logger.logToStderr(`Retrieving users ids...`);
+      }
+      return entraUser.getUserIdsByUpns(formatting.splitAndTrim(userNames));
+    }
+
+    return [];
+  }
+
+  private async updateUsers(groupId: string, role: 'members' | 'owners', userIds: string[]): Promise<void> {
+    const groupUsers = await odata.getAllItems<User>(`${this.resource}/v1.0/groups/${groupId}/${role}/microsoft.graph.user?$select=id`);
+    const userIdsToAdd = userIds.filter(userId => !groupUsers.some(groupUser => groupUser.id === userId));
+    const userIdsToRemove = groupUsers.filter(groupUser => !userIds.some(userId => groupUser.id === userId));
```

Comment:
It's a bit odd that we call this `userIdsToRemove` while it are not user IDs. Lets either rename this variable, or make sure that only IDs are stored in here.

Reply:
Yes, it' confusing. I will work only with user ids.